spring-data-cassandra/src/test/java/org/springframework/data/cassandra/repository/support/BasicMapIdUnitTests.java

```diff
@@ -41,4 +55,471 @@ public void testMapConstructor() {
 		assertThat(map.get("field1")).isEqualTo(basicMapId.get("field1"));
 		assertThat(map.get("field2")).isEqualTo(basicMapId.get("field2"));
 	}
+
+
+	@Test
+	public void testOne() throws Exception {
+
+		BasicMapId basicMapId = new BasicMapId();
+		basicMapId.putAll(basicMapId);
+
+		assertEquals(0, basicMapId.size());
+		assertTrue(basicMapId.isEmpty());
+	}
+
+
+	@Test
+	public void testTwo() throws Exception {
+
+		BasicMapId basicMapId = new BasicMapId();
+		int intOne = basicMapId.size();
+
+		assertEquals(0, basicMapId.size());
+		assertTrue(basicMapId.isEmpty());
+
+		assertEquals(0, intOne);
+	}
+
+
+	@Test
+	public void testThree() throws Exception {
+
+		BasicMapId basicMapId = (BasicMapId) BasicMapId.id("#xAD]-}", (Serializable) "#xAD]-}");
```

Comment:
I have made very, very good experiences providing not awaited, uncommon strings to systems under test! Especially those with special meaning! For example if you think of security breaches and the long time in the future they are supposed to run.
However, I've replaced it with "a".